dashboard/modules/job/job_head.py

```diff
@@ -31,6 +39,52 @@ def __init__(self, dashboard_head):
         # JobInfoGcsServiceStub
         self._gcs_job_info_stub = None
 
+    async def _next_job_id(self):
+        counter_str = await self._dashboard_head.aioredis_client.incr(
+            job_consts.REDIS_KEY_JOB_COUNTER)
+        job_id_int = int(counter_str)
+        return ray.JobID.from_int(job_id_int)
+
+    @routes.post("/jobs")
+    async def submit_job(self, req) -> aiohttp.web.Response:
+        agent_addresses = list(DataOrganizer.iter_agent_addresses())
+        if len(agent_addresses) == 0:
+            return dashboard_utils.rest_response(
+                success=False,
+                message=f"Failed to submit job: no nodes available.")
+
+        job_description_data = dict(await req.json())
+        # Validate the job description data.
+        try:
+            JobDescription(**job_description_data)
+        except Exception as ex:
+            return dashboard_utils.rest_response(
+                success=False, message=f"Failed to submit job: {ex}")
+        job_id = await self._next_job_id()
+
+        # Choose a random agent to start the driver for this job.
+        # TODO(fyrestone): The InitializeJobEnv request is now only used
+        # to start the driver process. We should send InitializeJobEnv requests
+        # to all agents to initialize the job environment on all nodes.
+        agent_address = random.choice(agent_addresses)
```

Comment:
This agent load balancing is adding quite a bit of complexity, do you think we could start by just always starting the driver on the head node?

Reply:
We do not assume there is a head node, because the GCS may be deployed separately without the raylet process in large cluster.

Reply:
I see... this deployment mode is not something that's supported in open source at all right now. Is this a P0 for this initial version?

Reply:
Another problem is that if we start all the drivers to the same node and the node goes down, then all the drivers will exit.

Reply:
Sure, but if the head node goes down the entire cluster goes down right now ;)

Reply:
We should keep the cluster running even the head node goes down, just can't start a new job.
- Deploy GCS and dashboard without raylet and redis.
- Use an external redis service.
- Discover nodes when the GCS is restarted.